mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -546,10 +574,88 @@ object KMeans {
       .run(data)
   }
 
+  private[spark] def validateInitMode(initMode: String): Boolean = {
+    initMode match {
+      case KMeans.RANDOM => true
+      case KMeans.K_MEANS_PARALLEL => true
+      case _ => false
+    }
+  }
+  private[spark] def validateDistanceMeasure(distanceMeasure: String): Boolean = {
+    distanceMeasure match {
+      case DistanceSuite.EUCLIDEAN => true
```

Comment:
You can use two labels in one statement if the result is the same; might be clearer. Match is probably overkill anyway

Reply:
I just wanted to be consistent with the similar implementation which is three lines above. Doing the same thing in two different ways a few lines of code after might be very confusing IMHO.